### ISSUE TITLE
CI: K8sServicesTest Checks service across nodes with L4 policy Tests NodePort with L4 Policy

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#11867]
```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/1.14-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:512
Request from testclient-4wxwt pod to service tftp://[::ffff:192.168.36.11]:30747/hello failed
Expected command: kubectl exec -n default testclient-4wxwt -- /bin/sh -c 'set -e; for i in $(seq 1 10); do curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 20 tftp://[::ffff:192.168.36.11]:30747/hello; done' 
To succeed, but it failed:
Exitcode: 28 
Stdout:
	 Hostname: testds-xfkdm
	 
	 Request Information:
	 	client_address=10.10.0.106
	 	client_port=59279
	 	real path=/hello
	 	request_scheme=tftp
	 
	 
	 Hostname: testds-2zkmq
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=51920
	 	real path=/hello
	 	request_scheme=tftp
	 
	 
	 Hostname: testds-2zkmq
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=48402
	 	real path=/hello
	 	request_scheme=tftp
	 
	 
	 Hostname: testds-2zkmq
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=38149
	 	real path=/hello
	 	request_scheme=tftp
	 
	 
	 Hostname: testds-2zkmq
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=38765
	 	real path=/hello
	 	request_scheme=tftp
	 
	 
	 Hostname: testds-xfkdm
	 
	 Request Information:
	 	client_address=10.10.0.106
	 	client_port=58164
	 	real path=/hello
	 	request_scheme=tftp
	 
	 
	 Hostname: testds-2zkmq
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=53457
	 	real path=/hello
	 	request_scheme=tftp
```

Assigning high priority because this was hit in the `test-missed-k8s` and those jobs are rarely executed so it might be a real bug and not a flake.
[3f7ee90a_K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy.zip](https://github.com/cilium/cilium/files/4724777/3f7ee90a_K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy.zip)

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/3151/testReport/junit/Suite-k8s-1/14/K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy/

**Comment by jrajahalme** [comment]
#11765 adds more debugging info in case this flakes again.

**Comment by jrajahalme** [comment]
#11819 (merged to master after this flake was seen) may affect this behavior.

**Comment by jrajahalme** [comment]
Its curious that 7 out of 10 tries succeed, and then the 8th fails. Both backends have served traffic so they are both functional. #11765 changes the test so that all 10 rounds are tried so that we'd know if the 9th and 10th round would succeed in a case like this.

**Comment by nebril** [comment]
Last fail in which this happened: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5170/testReport/junit/Suite-k8s-1/18/K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy/

[926078a9_K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy.zip](https://github.com/cilium/cilium/files/4885699/926078a9_K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy.zip)

**Comment by jrajahalme** [comment]
> Last fail in which this happened: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/5170/testReport/junit/Suite-k8s-1/18/K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy/
> 
> [926078a9_K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy.zip](https://github.com/cilium/cilium/files/4885699/926078a9_K8sServicesTest_Checks_service_across_nodes_with_L4_policy_Tests_NodePort_with_L4_Policy.zip)

This build was started before the fix in #12248 went in.

**Comment by jrajahalme** [comment]
Fixed by #12248
